src/Android/Avalonia.Android/AvaloniaActivity.cs

```diff
@@ -1,62 +1,42 @@
+using System;
+using Android.App;
+using Android.Content;
+using Android.Content.Res;
 using Android.OS;
+using Android.Runtime;
 using AndroidX.AppCompat.App;
-using Android.Content.Res;
 using AndroidX.Lifecycle;
-using Avalonia.Controls.ApplicationLifetimes;
-using Avalonia.Controls;
-using Android.Runtime;
-using Android.App;
-using Android.Content;
-using System;
 
 namespace Avalonia.Android
 {
     public abstract class AvaloniaActivity : AppCompatActivity
     {
-        internal class SingleViewLifetime : ISingleViewApplicationLifetime
-        {
-            public AvaloniaView View { get; internal set; }
-
-            public Control MainView
-            {
-                get => (Control)View.Content;
-                set => View.Content = value;
-            }
-        }
-
         internal Action<int, Result, Intent> ActivityResult;
         internal AvaloniaView View;
         internal AvaloniaViewModel _viewModel;
 
-        protected abstract AppBuilder CreateAppBuilder();
-
         protected override void OnCreate(Bundle savedInstanceState)
         {
-            var builder = CreateAppBuilder();
+            _viewModel = new ViewModelProvider(this).Get(Java.Lang.Class.FromType(typeof(AvaloniaViewModel))) as AvaloniaViewModel;
```

Comment:
Why do we need a view model at all? Can't we just use a static variable? There is only one App and two single instances of splash and main activity anyway.

Reply:
No. A static variable will have the same effect. I'll switch to that.